salesagility/SuiteCRM
# Acceptance Tests for basic module (Sugar Objects)

Task: SCRM-660

## Description



This change adds a new framework to help build acceptance tests for SuiteCRM. In this pull request I have added some page objects to represent the basic sugar object and the CSS breakpoints. I have added different step objects, to provide common steps which can be reused in tests. This includes using the navigation bar, the sidebar, the common views and creating modules in module builder. These can harness dependency injection, so that you can include them in the arguments of your test methods in a cest file.

This change includes different methods for running in different test environments.  I have also added more information to the tests/README.md file.

There is also functionality in the acceptance tester to login to suitecrm.

Plus i have some functionality which helps to provide conditional steps. See \Helper\Acceptance::seePageHas for more information.

I am use the  [faker](https://github.com/fzaninotto/Faker) package to mock data.

This doesn't yet provide full coverage of the basic module, instead we are covering just the high priority must haves.

## Motivation and Context

This provides automated tests:
- create basic module in module builder
- view basic module
- create record with basic module
- filter and select record
- edit record
- duplicate record
- delete record

**Note this does not yet test the list view features, and some actions like find duplicates and view change log**

## How To Test This

* See [tests/README.md](https://github.com/daniel-samson/SuiteCRM/blob/feature_codeception_basic_module/tests/README.md) and configure your environment appropriately
* Run `composer install`
* Run `codecept run install`
* Run `codecept run tests/acceptance/Core/BasicModuleCest.php`

## Types of changes


### Final checklist
